great_expectations/rule_based_profiler/types/attributed_resolved_metrics.py

```diff
@@ -10,6 +10,50 @@
 from great_expectations.types.attributes import Attributes
 
 
+def _condition_metric_values(metric_values: MetricValues) -> MetricValues:
+    def _detect_illegal_array_type_or_shape(values: MetricValues) -> bool:
+        # Python "None" is illegal as candidate for conversion into "numpy.ndarray" type.
+        if values is None:
+            return True
+
+        # Pandas "DataFrame" and "Series" are illegal as candidates for conversion into "numpy.ndarray" type.
+        if isinstance(values, (pd.DataFrame, pd.Series, set)):
+            return True
+
+        if isinstance(values, (list, tuple)):
+            value: MetricValue = values[0]
+            # Python "None" is illegal as candidate for conversion into "numpy.ndarray" type.
+            if value is None:
+                return True
+
+            # Pandas "DataFrame" and "Series" are illegal as candidates for conversion into "numpy.ndarray" type.
+            if isinstance(value, (pd.DataFrame, pd.Series, set)):
+                return True
```

Comment:
What's the reason behind checking `values[0]`? Aren't we performing this check by recursively calling the func on each element down in L46?

Reply:
@cdkini This is just a way to check one of the elements to see what the type is.  If there is a more elegant way, I would be curious to know -- thanks!